src/MaIN.Domain/Entities/Agents/AgentSource/AgentAPISourceDetails.cs

```diff
@@ -8,4 +8,9 @@ public class AgentApiSourceDetails : AgentSourceDetailsBase, IAgentSource
     public string? Query { get; set; }
     public string? ResponseType { get; init; }
     public int? ChunkLimit { get; init; }
+    public string? AuthorisationToken { get; set; }
```

Comment:
If it will end up with bearer token, use BearerToken as name, If its more of ApiKey convention then. "ApiKey" AuthorisationToken contains typo, and if we want to stick to this pattern AuthenticationToken would be more suitable. But I personally would vote for option 1/2